### DISCUSSION TITLE
Drag is laggy

### DISCUSSION THREAD

**Discussion by khyati18** [sceneview/sceneview#288] [General]

**Comment by ThomasGorisse** [comment]
Drag lag in older SceneView versions has been improved in v4.0.0-rc.1. The Compose-based architecture and updated Filament renderer provide smoother gesture handling.

If you still experience lag, common causes:
1. **Model complexity** — High poly models slow down hit testing during drag
2. **Many nodes** — Collision detection iterates through all nodes
3. **Background work** — Ensure model loading isn't happening on the main thread (use `rememberModelInstance` which handles this)

Upgrade to v4.0.0-rc.1 (`io.github.sceneview:sceneview:4.0.0-rc.1`) and check if the issue persists.
